### PR TITLE
google auth and openapi

### PR DESCRIPTION
**Pull request by hrvojevincek**

### CODE REVIEW

**Review by hrvojevincek — COMMENTED** [review]
minor conflicts in the files.

**Review by arol — COMMENTED** [review]
It's a single PR for two main features. Next time create a single PR for each of them.

File: src/app/api/GPTClient.ts [hunk 1]
```diff
@@ -16,15 +21,30 @@ class MockGPTStrategy implements GPTClientStrategy {
...
+      model: "text-davinci-003",
+      prompt: "say hello",
+      temperature: 0,
+      max_tokens: 849,
+      top_p: 1,
+      frequency_penalty: 0,
+      presence_penalty: 0,
+    });
+    //if (userInfo) connect to user table
+    console.log(response.data);
```
Comment (arol) [inline src/app/api/GPTClient.ts:45]: Mock strategy should not talk to openai.

File: src/app/api/GPTClient.ts [hunk 2]
```diff
@@ -34,13 +54,12 @@ class RealGPTStrategy implements GPTClientStrategy {
...
-          duration: parseInt(duration),
+        destination: destination.toLowerCase(),
+        duration: parseInt(duration),
       },
     });
     if (search?.response) {
       const data = JSON.parse(search?.response);
-      console.log(data, "CLIENT");
       return data;
     }
```
Comment (arol) [inline src/app/api/GPTClient.ts:64]: I think you messed up the real and the mock strategy.

**Review by arol — CHANGES_REQUESTED** [review]
It's a single PR for two main features. Next time create a single PR for each of them.

**Pull request merged**